### PR TITLE
PP-4463 Server-side validation for responsible person page

### PR DESCRIPTION
**Pull request by alexbishop1**
Except for date of birth validation, which is coming later.

Update max lengths to match what we think Stripe accepts.

with @cobainc0

### CODE REVIEW

File: app/controllers/stripe-setup/responsible-person/responsible-person-validations.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+'use strict'
```
Comment (sfount) [inline app/controllers/stripe-setup/responsible-person/responsible-person-validations.test.js]: Do we name any other tests `.test.js`? I haven't seen this before - I think normally `spec` is used.

File: app/controllers/stripe-setup/responsible-person/post.controller.js [hunk 2]
```diff
@@ -2,27 +2,98 @@
...
+    maxLength: 100
+  },
+  {
+    field: HOME_ADDRESS_POSTCODE_FIELD,
+    validator: validatePostcode
+  }
+]
 
 module.exports = (req, res) => {
+  const errors = validationRules.reduce((errors, validationRule) => {
```
Comment (sfount) [inline app/controllers/stripe-setup/responsible-person/post.controller.js]: 👍

File: app/controllers/stripe-setup/responsible-person/post.controller.js [hunk 3]
```diff
@@ -2,27 +2,98 @@
...
+  } else if (lodash.get(req.body, 'answers-need-changing') === 'true') {
     return response.response(req, res, 'stripe-setup/responsible-person/index', pageData)
+  } else {
+    return response.response(req, res, 'stripe-setup/responsible-person/check-your-answers', pageData)
   }
-  return response.response(req, res, 'stripe-setup/responsible-person/check-your-answers', pageData)
+}
+
+const validate = (req, fieldName, fieldValidator, maxLength) => {
+  const field = getFieldValue(req, fieldName)
```
Comment (sfount) [inline app/controllers/stripe-setup/responsible-person/post.controller.js]: Is there a need for this method? It looks like it directly maps to `lodash.get`

File: app/controllers/stripe-setup/responsible-person/responsible-person-validations.test.js (hunk 1 above)
Comment (alexbishop1) [reply app/controllers/stripe-setup/responsible-person/responsible-person-validations.test.js]: We have … a few different conventions, including `.test.js`

File: app/controllers/stripe-setup/responsible-person/post.controller.js (hunk 3 above)
Comment (alexbishop1) [reply app/controllers/stripe-setup/responsible-person/post.controller.js]: We’ll remove this in a future PR

**Pull request merged**